api/application/client.go

```diff
@@ -65,11 +65,45 @@ func (c *Client) ModelUUID() string {
 	return tag.Id()
 }
 
+// CharmOriginSource represents the source of the charm.
+type CharmOriginSource string
+
+func (c CharmOriginSource) String() string {
+	return string(c)
+}
+
+const (
+	// OriginLocal represents a local charm.
+	OriginLocal CharmOriginSource = "local"
+	// OriginCharmStore represents a charm from the now old charmstore.
+	OriginCharmStore CharmOriginSource = "charm-store"
+	// OriginCharmhub represents a charm from the new charmhub.
+	OriginCharmhub CharmOriginSource = "charmhub"
+	// OriginUnknown represents that we don't know where this charm came from.
+	// Either the charm was migrated up from an older version of Juju or we
+	//  didn't have enough information when we set the charm.
+	OriginUnknown CharmOriginSource = "unknown"
```

Comment:
Is this possible?  If it doesn't have "cs:" or "local:", then it's charm-hub.

Reply:
Once we flesh this out for deployment and upgrading, unknown should be only a source of errors.